cogs/afterhours/afterhours.py

```diff
@@ -265,3 +420,245 @@ async def afterHoursSet(self, ctx: Context):
+                await ctx.send(
+                    f"The inactive duration for auto-purge is currently set to {AfterHours.strAutoPurgeInactiveDuration(*await self.getAutoPurgeInactiveDuration(ctx.guild))}."
+                )
+
+    async def getAutoPurgeInactiveDuration(self, guild: discord.Guild) -> List[int]:
+        """Get the inactive duration for auto-purge.
+
+        Returns
+        -------
+        Returns a list of the values for each time unit, in the following order:
+        - Number of years.
+        - Number of months.
+        - Number of weeks.
+        - Number of days.
+        - Number of hours.
+        - Number of minutes.
+        - Number of seconds.
+        """
+        inactiveDurationConfig = await self.config.guild(guild).get_raw(
+            KEY_AUTO_PURGE, KEY_INACTIVE_DURATION
+        )
+        return [
+            int(x) if x else 0
+            for x in [
+                inactiveDurationConfig[key]
+                for key in [
+                    KEY_INACTIVE_DURATION_YEARS,
+                    KEY_INACTIVE_DURATION_MONTHS,
+                    KEY_INACTIVE_DURATION_WEEKS,
+                    KEY_INACTIVE_DURATION_DAYS,
+                    KEY_INACTIVE_DURATION_HOURS,
+                    KEY_INACTIVE_DURATION_MINUTES,
+                    KEY_INACTIVE_DURATION_SECONDS,
+                ]
+            ]
+        ]
+
+    async def setAutoPurgeInactiveDuration(
+        self,
+        guild: discord.Guild,
+        years: int = 0,
+        months: int = 0,
+        weeks: int = 0,
+        days: int = 0,
+        hours: int = 0,
+        minutes: int = 0,
+        seconds: int = 0,
+    ) -> None:
+        """Set the inactive duration for auto-purge.
+
+        Parameters
+        ----------
+        guild: discord.Guild
+            The guild to set the inactive duration for.
+        years: int
+            The number of years.
+        months: int
+            The number of months.
+        weeks: int
+            The number of weeks.
+        days: int
+            The number of days.
+        hours: int
+            The number of hours.
+        minutes: int
+            The number of minutes.
+        seconds: int
+            The number of seconds.
+        """
+        async with self.config.guild(guild).get_attr(KEY_AUTO_PURGE).get_attr(
+            KEY_INACTIVE_DURATION
+        )() as inactiveDurationConfig:
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_YEARS] = years
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_MONTHS] = months
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_WEEKS] = weeks
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_DAYS] = days
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_HOURS] = hours
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_MINUTES] = minutes
+            inactiveDurationConfig[KEY_INACTIVE_DURATION_SECONDS] = seconds
+
+    @staticmethod
+    def strAutoPurgeInactiveDuration(
```

Comment:
Can you see if `redbot.core.utils.chat_formatting.humanize_timedelta` achieves the same or similar result that you're looking for.

Reply:
`redbot.core.utils.chat_formatting.humanize_timedelta` is perfect.